### PR TITLE
M3-1393 refactor account

### PR DESCRIPTION
**Pull request by Jskobos**
* Add documentation for account endpoint-related things
* Add validation schemas as appropriate, updating error handling when necessary.
* Standardized function signatures, moved filters from `services` and into the Manager proper

### CODE REVIEW

File: src/services/account/payments.ts [hunk 1]
```diff
@@ -24,43 +30,95 @@ interface PaymentID {
...
+ * @param data.ccv { string } the 3-digit code on the back of the
+ * credit card.
+ * 
+ */
 export const makePayment = (data: { usd: string, ccv: string }) =>
   Request<Linode.Payment>(
     setURL(`${API_ROOT}/account/payments`),
     setMethod('POST'),
-    setData(data),
+    setData(data, PaymentSchema),
```
Comment (martinmckenna) [inline src/services/account/payments.ts]: Now that we have this, we can get rid of lines 180-185 of `MakeAPaymentPanel.tsx`

File: src/services/account/account.schema.ts [hunk 2]
```diff
@@ -42,4 +42,74 @@ export const updateOAuthClientSchema = object({
...
+
+export const CreditCardSchema = object({
+  card_number: string()
+    .required("Credit card number is required.")
+    .min(13, "Credit card number must be between 13 and 23 characters.")
+    .max(23, "Credit card number must be between 13 and 23 characters."),
+  expiry_year: number()
+    .required("Expiration year is required.")
+    .min(1970, "Expiration year must be four digits.")
+    .max(2050, "Expiration year must be four digits."),
```
Comment (martinmckenna) [inline src/services/account/account.schema.ts]: Let's not hardcode these numbers and instead base it from `new Date().getFullYear()` or something

File: src/features/Account/AccountPanels/MakeAPaymentPanel/MakeAPaymentPanel.tsx [hunk 3]
```diff
@@ -432,7 +432,7 @@ class MakeAPaymentPanel extends React.Component<CombinedProps, State> {
                 onChange={this.handleCVVChange}
                 value={this.state.CVV}
                 required
-                type="number"
+                type="text"
```
Comment (martinmckenna) [inline src/features/Account/AccountPanels/MakeAPaymentPanel/MakeAPaymentPanel.tsx]: what's the point of this change? Can CVV ever have letters?

Comment (patthiel) [reply src/features/Account/AccountPanels/MakeAPaymentPanel/MakeAPaymentPanel.tsx]: There are issues with CVV's that start with `0` when the type is number. On firefox, it displays with these arrow indicators to increment it. (See screenshot). It solves a weird edge case where, if a user enters a CVV, toggles Paypal and toggles back the CVV changes from say. `044` to `44` 
screen shot 2018-10-26 at 9 58 56 am

**Pull request merged**